File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack=[]
        for ch in tokens:
            if ch in "+-/*":
                pop1=stack.pop()
                pop2=stack.pop()
                if ch=='+':
                    res=pop2+pop1
                elif ch=='-':
                    res=pop2-pop1
                elif ch=='*':
                    res=pop2*pop1
                elif ch=='/':
                    res=int(eval(str(pop2)+"/"+str(pop1)))
                stack.append(res)
            else:
                stack.append(int(ch))
        return stack[-1]

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        
        Stack<Integer> stack=new Stack<>();

        for(String ch: tokens){
            int res=0;
            if(ch.equals("+")==true){
                int num1=stack.pop();
                int num2=stack.pop();
                res=num2+num1;
            }
            else if(ch.equals("-")==true){
                int num1=stack.pop();
                int num2=stack.pop();
                res=num2-num1;
            }
            else if(ch.equals("*")==true){
                int num1=stack.pop();
                int num2=stack.pop();
                res=num2*num1;
            }
            else if(ch.equals("/")==true){
                int num1=stack.pop();
                int num2=stack.pop();
                res=num2/num1;
            }
           else{
             res=Integer.parseInt(ch);
           }

           stack.push(res);
            } 



            return stack.pop();



        
    }
}